Sources/SwiftDocC/Model/Rendering/References/ExternalLocationReference.swift

```diff
@@ -0,0 +1,53 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+/// A specialized ``DownloadReference`` used for references to external links.
+///
+/// `@CallToAction` directives can link either to a local file or to a URL, whether relative or
+/// absolute. Directives that use the `file` argument will create a ``DownloadReference`` and copy
+/// the file from the catalog into the resulting archive.
+///
+/// An `ExternalLocationReference` is intended to encode to Render JSON compatible with a
+/// ``DownloadReference``, but with the `url` set to the text given in the `@CallToAction`'s `url`
+/// argument.
+public struct ExternalLocationReference: RenderReference, URLReference {
+    public static var baseURL: URL = .init(string: "/\(DownloadReference.locationName)/")!
+
+    public var type: RenderReferenceType = .download
```

Comment:
Should we require this to be RenderReferenceType.download?

```suggestion
    public private(set) var type: RenderReferenceType = .download
```